aiida_optimade/translators/entities.py

```diff
@@ -154,20 +154,17 @@ def _store_attributes_node_extra(self) -> None:
 
         profile = get_manager().get_profile()
         LOGGER.debug(
-            "(%s) Updating Node %s in AiiDA DB!", profile.database_backend, self._pk
+            "(%s) Updating Node %s in AiiDA DB!", profile.storage_backend, self._pk
         )
-        if profile.database_backend == "django":
-            from aiida.backends.djsite.db.models import DbNode
-
-            with get_manager().get_backend().transaction():
-                DbNode.objects.filter(pk=self._pk).update(extras=extras)
-        elif profile.database_backend == "sqlalchemy":
-            from aiida.backends.sqlalchemy.models.node import DbNode
+        if profile.storage_backend == "psql_dos":
+            # TODO: this can be `set_extra` directly? is there performance issue with sqlite_zip?
+            from aiida.storage.psql_dos.models.node import DbNode
 
             with get_manager().get_backend().transaction() as session:
                 session.query(DbNode).filter(DbNode.id == self._pk).update(
                     values={"extras": extras}
```

Comment:
To be honest I am not sure anymore how much going to our front-end slows down these queries. It would be good to test since it has been refactored a lot and the slowdown may not be so bad anymore compared to before. If that is the case, I would definitely advise going through the front-end such that all storage backends are automatically supported, especially now that they are pluginnable with aiida v2.0.

Reply:
That's fair enough, and some benchmarking would be more than welcome. I'm pretty sure this will come out to still be faster, as the Node is never (supposedly) retrieved from the database, but rather a specific DB row's `extras` column is updated.